homeassistant/components/sensor/xs1.py

```diff
@@ -0,0 +1,57 @@
+"""
+Support for XS1 sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.xs1/
+"""
+
+import logging
+
+from homeassistant.components.xs1 import (
+    ACTUATORS, DOMAIN as COMPONENT_DOMAIN, SENSORS, XS1DeviceEntity)
+from homeassistant.helpers.entity import Entity
+
+DEPENDENCIES = ['xs1']
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the XS1 sensor platform."""
+    from xs1_api_client.api_constants import ActuatorType
+
+    sensors = hass.data[COMPONENT_DOMAIN][SENSORS]
+    actuators = hass.data[COMPONENT_DOMAIN][ACTUATORS]
+
+    sensor_entities = []
+    for sensor in sensors:
+        belongs_to_climate_actuator = False
+        for actuator in actuators:
```

Comment:
Move this to a separate iteration and save the result. That will save iterations.
```py
saved_actuators = {
    actuator.name(): actuator.type()
    for actuator in actuators}
for sensor in sensors:
     if saved_actuators.get(
            sensor.name()) == ActuatorType.TEMPERATURE:
        continue
    sensor_entities.append(XS1Sensor(sensor))
```

Reply:
It doesn't work like that though, because the actuator and sensor names may not be equal (even though it should be possible in theory **the gateway does not allow this**)
Because of this I used the "in" notation.

Reply:
The worst case of this loop is 64 * 64 which to be fair is quite a lot, but since only configured and active devices (on the gateway side) are considered in this loop I highly doubt this number will actually be reached by anyone. Additionally this check will only happen once when hass is started (at least in the current implementation).

Reply:
Ok, :+1:.